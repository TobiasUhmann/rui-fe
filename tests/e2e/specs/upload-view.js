describe('Upload Page', () => {

    it('Load page', () => {

        // WHEN visiting the "Upload" page
        // THEN the page should be shown

        cy.visit('/upload')

        // TODO
    })

    it('Load page via root URL', () => {

        // WHEN visiting the "Upload" page via the root URL
        // THEN the page should be shown

        cy.visit('/')

        cy.url().should('contain', '/upload')

        // TODO
    })

    it('Cancel upload', () => {

        // WHEN clicking the "Choose File" button
        // AND  selecting a valid "Symptax Upload ZIP"
        // AND  clicking the "Upload" button
        // AND  clicking the "Cancel" button in the opening confirmation dialog
        // THEN the user should stay on the page
        // AND  the file selection should still contain the file

        cy.get('input[type="file"]').attachFile('symptax_upload_v7_random.zip')

        cy.get('[type="submit"]').click()
        cy.get('.cancel').click()

        // TODO
    })

    it('Upload ZIP', () => {

        // WHEN clicking the "Upload" button again
        // AND  clicking the "Overwrite" button
        // THEN the "Taxonomy" page should be shown
        // AND  the uploaded ZIP's contents should be shown

        cy.get('[type="submit"]').click()
        cy.get('.continue').click()

        cy.url().should('contain', '/taxonomy')
        cy.get('.node-name').should('contain', 'C-1')

        // TODO
    })
})
