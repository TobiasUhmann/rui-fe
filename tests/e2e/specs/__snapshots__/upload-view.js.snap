exports[`Upload Page > toMatchSnapshot - HTML #0`] = `
<body>
  <noscript
    ><strong
      >We're sorry but rui-fe doesn't work properly without JavaScript enabled.
      Please enable it to continue.</strong
    ></noscript
  >
  <div id="app" data-v-app="">
    <div class="app">
      <header data-v-47d8dfc0="">
        <div class="title" data-v-47d8dfc0="">
          <h1 data-v-47d8dfc0="">RUI</h1>
          <h2 data-v-47d8dfc0="">Hunting<br data-v-47d8dfc0="" />Entities</h2>
        </div>
        <nav data-v-47d8dfc0="">
          <a
            href="/upload"
            class="router-link-active router-link-exact-active"
            data-v-47d8dfc0=""
            aria-current="page"
            >Upload</a
          ><a href="/taxonomy" class="" data-v-47d8dfc0="">Taxonomy</a>
        </nav>
        <div class="logo" data-v-47d8dfc0="">
          <img
            src="/img/logo-lavis.2cfaf679.svg"
            alt="LAVIS Logo"
            data-v-47d8dfc0=""
          />
        </div>
      </header>
      <main>
        <div
          class="modal-background d-none"
          data-v-6020d849=""
          data-v-8d0f9738=""
        >
          <div class="modal" data-v-6020d849="">
            <header class="modal-header" data-v-6020d849="">
              <h1 data-v-6020d849="">Caution</h1>
              <div class="close-button" data-v-6020d849="">×</div>
            </header>
            <main class="modal-content" data-v-6020d849="">
              <p class="message" data-v-6020d849="">
                Uploading will <b data-v-6020d849="">overwrite</b> the existing
                taxonomy and matches!
              </p>
              <div class="buttons" data-v-6020d849="">
                <button class="button cancel" data-v-6020d849="">Cancel</button
                ><button class="button continue" data-v-6020d849="">
                  Overwrite
                </button>
              </div>
            </main>
          </div>
        </div>
        <form class="upload-form" data-v-8d0f9738="">
          <label for="symptaxUploadZip" data-v-8d0f9738=""
            >Symptax Upload ZIP</label
          ><input
            id="symptaxUploadZip"
            class="upload-form-input"
            type="file"
            name="symptaxUploadZip"
            data-v-8d0f9738=""
          /><input
            class="upload-form-input"
            type="submit"
            value="Upload"
            data-v-8d0f9738=""
          />
        </form>
      </main>
      <footer data-v-a28185ec="">
        <img
          src="/img/logo-empolis.0a2277d8.png"
          alt="Empolis Logo"
          data-v-a28185ec=""
        /><img
          src="/img/logo-hsrm.2a007289.svg"
          alt="HSRM Logo"
          data-v-a28185ec=""
        />
      </footer>
    </div>
  </div>
  <script src="/js/chunk-vendors.8faaf858.js"></script>
  <script src="/js/app.9d6f2ea9.js"></script>
</body>
`;

exports[`Upload Page > Load page #0`] = `
<html lang="">
  <head>
    <script type="text/javascript">
      document.domain = "localhost";
      var Cypress = (window.Cypress = parent.Cypress);
      if (!Cypress) {
        throw new Error(
          "Something went terribly wrong and we cannot proceed. We expected to find the global Cypress in the parent window but it is missing!. This should never happen and likely is a bug. Please open an issue!"
        );
      }
      Cypress.action("app:window:before:load", window);
    </script>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <link rel="icon" href="/favicon.ico" />
    <title>rui-fe</title>
    <link href="/css/predictions.6b9f84fd.css" rel="prefetch" />
    <link href="/css/predictions~taxonomy.2f7104ec.css" rel="prefetch" />
    <link href="/css/taxonomy.ab43e5fe.css" rel="prefetch" />
    <link href="/css/upload.30e252e9.css" rel="prefetch" />
    <link href="/js/predictions.7485ee49.js" rel="prefetch" />
    <link href="/js/predictions~taxonomy.48b0d5a4.js" rel="prefetch" />
    <link href="/js/taxonomy.07a18798.js" rel="prefetch" />
    <link href="/js/upload.ae3f84e4.js" rel="prefetch" />
    <link href="/css/app.e7f63673.css" rel="preload" as="style" />
    <link href="/js/app.9d6f2ea9.js" rel="preload" as="script" />
    <link href="/js/chunk-vendors.8faaf858.js" rel="preload" as="script" />
    <link href="/css/app.e7f63673.css" rel="stylesheet" />
    <link rel="stylesheet" type="text/css" href="/css/upload.30e252e9.css" />
    <script charset="utf-8" src="/js/upload.ae3f84e4.js"></script>
  </head>
  <body>
    <noscript
      ><strong
        >We're sorry but rui-fe doesn't work properly without JavaScript
        enabled. Please enable it to continue.</strong
      ></noscript
    >
    <div id="app" data-v-app="">
      <div class="app">
        <header data-v-47d8dfc0="">
          <div class="title" data-v-47d8dfc0="">
            <h1 data-v-47d8dfc0="">RUI</h1>
            <h2 data-v-47d8dfc0="">Hunting<br data-v-47d8dfc0="" />Entities</h2>
          </div>
          <nav data-v-47d8dfc0="">
            <a
              href="/upload"
              class="router-link-active router-link-exact-active"
              data-v-47d8dfc0=""
              aria-current="page"
              >Upload</a
            ><a href="/taxonomy" class="" data-v-47d8dfc0="">Taxonomy</a>
          </nav>
          <div class="logo" data-v-47d8dfc0="">
            <img
              src="/img/logo-lavis.2cfaf679.svg"
              alt="LAVIS Logo"
              data-v-47d8dfc0=""
            />
          </div>
        </header>
        <main>
          <div
            class="modal-background d-none"
            data-v-6020d849=""
            data-v-8d0f9738=""
          >
            <div class="modal" data-v-6020d849="">
              <header class="modal-header" data-v-6020d849="">
                <h1 data-v-6020d849="">Caution</h1>
                <div class="close-button" data-v-6020d849="">×</div>
              </header>
              <main class="modal-content" data-v-6020d849="">
                <p class="message" data-v-6020d849="">
                  Uploading will <b data-v-6020d849="">overwrite</b> the
                  existing taxonomy and matches!
                </p>
                <div class="buttons" data-v-6020d849="">
                  <button class="button cancel" data-v-6020d849="">
                    Cancel</button
                  ><button class="button continue" data-v-6020d849="">
                    Overwrite
                  </button>
                </div>
              </main>
            </div>
          </div>
          <form class="upload-form" data-v-8d0f9738="">
            <label for="symptaxUploadZip" data-v-8d0f9738=""
              >Symptax Upload ZIP</label
            ><input
              id="symptaxUploadZip"
              class="upload-form-input"
              type="file"
              name="symptaxUploadZip"
              data-v-8d0f9738=""
            /><input
              class="upload-form-input"
              type="submit"
              value="Upload"
              data-v-8d0f9738=""
            />
          </form>
        </main>
        <footer data-v-a28185ec="">
          <img
            src="/img/logo-empolis.0a2277d8.png"
            alt="Empolis Logo"
            data-v-a28185ec=""
          /><img
            src="/img/logo-hsrm.2a007289.svg"
            alt="HSRM Logo"
            data-v-a28185ec=""
          />
        </footer>
      </div>
    </div>
    <script src="/js/chunk-vendors.8faaf858.js"></script>
    <script src="/js/app.9d6f2ea9.js"></script>
  </body>
</html>
`;

exports[`Upload Page > Load page via root URL #0`] = `
<html lang="">
  <head>
    <script type="text/javascript">
      document.domain = "localhost";
      var Cypress = (window.Cypress = parent.Cypress);
      if (!Cypress) {
        throw new Error(
          "Something went terribly wrong and we cannot proceed. We expected to find the global Cypress in the parent window but it is missing!. This should never happen and likely is a bug. Please open an issue!"
        );
      }
      Cypress.action("app:window:before:load", window);
    </script>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <link rel="icon" href="/favicon.ico" />
    <title>rui-fe</title>
    <link href="/css/predictions.6b9f84fd.css" rel="prefetch" />
    <link href="/css/predictions~taxonomy.2f7104ec.css" rel="prefetch" />
    <link href="/css/taxonomy.ab43e5fe.css" rel="prefetch" />
    <link href="/css/upload.30e252e9.css" rel="prefetch" />
    <link href="/js/predictions.7485ee49.js" rel="prefetch" />
    <link href="/js/predictions~taxonomy.48b0d5a4.js" rel="prefetch" />
    <link href="/js/taxonomy.07a18798.js" rel="prefetch" />
    <link href="/js/upload.ae3f84e4.js" rel="prefetch" />
    <link href="/css/app.e7f63673.css" rel="preload" as="style" />
    <link href="/js/app.9d6f2ea9.js" rel="preload" as="script" />
    <link href="/js/chunk-vendors.8faaf858.js" rel="preload" as="script" />
    <link href="/css/app.e7f63673.css" rel="stylesheet" />
    <link rel="stylesheet" type="text/css" href="/css/upload.30e252e9.css" />
    <script charset="utf-8" src="/js/upload.ae3f84e4.js"></script>
  </head>
  <body>
    <noscript
      ><strong
        >We're sorry but rui-fe doesn't work properly without JavaScript
        enabled. Please enable it to continue.</strong
      ></noscript
    >
    <div id="app" data-v-app="">
      <div class="app">
        <header data-v-47d8dfc0="">
          <div class="title" data-v-47d8dfc0="">
            <h1 data-v-47d8dfc0="">RUI</h1>
            <h2 data-v-47d8dfc0="">Hunting<br data-v-47d8dfc0="" />Entities</h2>
          </div>
          <nav data-v-47d8dfc0="">
            <a
              href="/upload"
              class="router-link-active router-link-exact-active"
              data-v-47d8dfc0=""
              aria-current="page"
              >Upload</a
            ><a href="/taxonomy" class="" data-v-47d8dfc0="">Taxonomy</a>
          </nav>
          <div class="logo" data-v-47d8dfc0="">
            <img
              src="/img/logo-lavis.2cfaf679.svg"
              alt="LAVIS Logo"
              data-v-47d8dfc0=""
            />
          </div>
        </header>
        <main>
          <div
            class="modal-background d-none"
            data-v-6020d849=""
            data-v-8d0f9738=""
          >
            <div class="modal" data-v-6020d849="">
              <header class="modal-header" data-v-6020d849="">
                <h1 data-v-6020d849="">Caution</h1>
                <div class="close-button" data-v-6020d849="">×</div>
              </header>
              <main class="modal-content" data-v-6020d849="">
                <p class="message" data-v-6020d849="">
                  Uploading will <b data-v-6020d849="">overwrite</b> the
                  existing taxonomy and matches!
                </p>
                <div class="buttons" data-v-6020d849="">
                  <button class="button cancel" data-v-6020d849="">
                    Cancel</button
                  ><button class="button continue" data-v-6020d849="">
                    Overwrite
                  </button>
                </div>
              </main>
            </div>
          </div>
          <form class="upload-form" data-v-8d0f9738="">
            <label for="symptaxUploadZip" data-v-8d0f9738=""
              >Symptax Upload ZIP</label
            ><input
              id="symptaxUploadZip"
              class="upload-form-input"
              type="file"
              name="symptaxUploadZip"
              data-v-8d0f9738=""
            /><input
              class="upload-form-input"
              type="submit"
              value="Upload"
              data-v-8d0f9738=""
            />
          </form>
        </main>
        <footer data-v-a28185ec="">
          <img
            src="/img/logo-empolis.0a2277d8.png"
            alt="Empolis Logo"
            data-v-a28185ec=""
          /><img
            src="/img/logo-hsrm.2a007289.svg"
            alt="HSRM Logo"
            data-v-a28185ec=""
          />
        </footer>
      </div>
    </div>
    <script src="/js/chunk-vendors.8faaf858.js"></script>
    <script src="/js/app.9d6f2ea9.js"></script>
  </body>
</html>
`;

exports[`Upload Page > Upload valid ZIP #0`] = `
<html lang="">
  <head>
    <script type="text/javascript">
      document.domain = "localhost";
      var Cypress = (window.Cypress = parent.Cypress);
      if (!Cypress) {
        throw new Error(
          "Something went terribly wrong and we cannot proceed. We expected to find the global Cypress in the parent window but it is missing!. This should never happen and likely is a bug. Please open an issue!"
        );
      }
      Cypress.action("app:window:before:load", window);
    </script>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <link rel="icon" href="/favicon.ico" />
    <title>rui-fe</title>
    <link href="/css/predictions.6b9f84fd.css" rel="prefetch" />
    <link href="/css/predictions~taxonomy.2f7104ec.css" rel="prefetch" />
    <link href="/css/taxonomy.ab43e5fe.css" rel="prefetch" />
    <link href="/css/upload.30e252e9.css" rel="prefetch" />
    <link href="/js/predictions.7485ee49.js" rel="prefetch" />
    <link href="/js/predictions~taxonomy.48b0d5a4.js" rel="prefetch" />
    <link href="/js/taxonomy.07a18798.js" rel="prefetch" />
    <link href="/js/upload.ae3f84e4.js" rel="prefetch" />
    <link href="/css/app.e7f63673.css" rel="preload" as="style" />
    <link href="/js/app.9d6f2ea9.js" rel="preload" as="script" />
    <link href="/js/chunk-vendors.8faaf858.js" rel="preload" as="script" />
    <link href="/css/app.e7f63673.css" rel="stylesheet" />
    <link rel="stylesheet" type="text/css" href="/css/upload.30e252e9.css" />
    <script charset="utf-8" src="/js/upload.ae3f84e4.js"></script>
    <link
      rel="stylesheet"
      type="text/css"
      href="/css/predictions~taxonomy.2f7104ec.css"
    />
    <script charset="utf-8" src="/js/predictions~taxonomy.48b0d5a4.js"></script>
    <link rel="stylesheet" type="text/css" href="/css/taxonomy.ab43e5fe.css" />
    <script charset="utf-8" src="/js/taxonomy.07a18798.js"></script>
  </head>
  <body>
    <noscript
      ><strong
        >We're sorry but rui-fe doesn't work properly without JavaScript
        enabled. Please enable it to continue.</strong
      ></noscript
    >
    <div id="app" data-v-app="">
      <div class="app">
        <header data-v-47d8dfc0="">
          <div class="title" data-v-47d8dfc0="">
            <h1 data-v-47d8dfc0="">RUI</h1>
            <h2 data-v-47d8dfc0="">Hunting<br data-v-47d8dfc0="" />Entities</h2>
          </div>
          <nav data-v-47d8dfc0="">
            <a href="/upload" class="" data-v-47d8dfc0="">Upload</a
            ><a
              href="/taxonomy"
              class="router-link-active router-link-exact-active"
              data-v-47d8dfc0=""
              aria-current="page"
              >Taxonomy</a
            >
          </nav>
          <div class="logo" data-v-47d8dfc0="">
            <img
              src="/img/logo-lavis.2cfaf679.svg"
              alt="LAVIS Logo"
              data-v-47d8dfc0=""
            />
          </div>
        </header>
        <main>
          <main class="grid" data-v-43bd4af4="">
            <section class="grid-section grid-left" data-v-43bd4af4="">
              <h1 class="grid-header" data-v-43bd4af4="">Taxonomy</h1>
              <div class="grid-content" data-v-25caf12c="" data-v-43bd4af4="">
                <ul data-v-25caf12c="">
                  <li class="extendable" data-v-31cb8309="" data-v-25caf12c="">
                    <span class="node-name" data-v-31cb8309="">A-1 (+0)</span
                    ><!---->
                  </li>
                  <li class="extendable" data-v-31cb8309="" data-v-25caf12c="">
                    <span class="node-name" data-v-31cb8309="">B-1 (+0)</span
                    ><!---->
                  </li>
                  <li class="" data-v-31cb8309="" data-v-25caf12c="">
                    <span class="node-name" data-v-31cb8309="">C-1 (+0)</span
                    ><!---->
                  </li>
                  <li data-v-25caf12c="">
                    <button class="create-button" data-v-25caf12c="">
                      Add Root Node
                    </button>
                  </li>
                </ul>
              </div>
            </section>
            <!----><!----><!---->
          </main>
        </main>
        <footer data-v-a28185ec="">
          <img
            src="/img/logo-empolis.0a2277d8.png"
            alt="Empolis Logo"
            data-v-a28185ec=""
          /><img
            src="/img/logo-hsrm.2a007289.svg"
            alt="HSRM Logo"
            data-v-a28185ec=""
          />
        </footer>
      </div>
    </div>
    <script src="/js/chunk-vendors.8faaf858.js"></script>
    <script src="/js/app.9d6f2ea9.js"></script>
  </body>
</html>
`;
